src/main/java/io/wispforest/lavender/client/LavenderBookScreen.java

```diff
@@ -608,9 +608,34 @@ protected List<FlowLayout> buildEntryIndex(Collection<Entry> entries, boolean re
                             if (!entryVisible) return;
 
                             var entryTitle = entry.title().toLowerCase(Locale.ROOT);
+
+                            var entryMatches = true;
                             for (var term : filter) {
-                                if (!entryTitle.contains(term)) return;
+                                if (!entryTitle.contains(term)) {
```

Comment:
If you may, I'd love to see this rewritten in a less imperative manner using streams and bulk collection methods like `containsAll`, which should condense this all into no more than a few lines.

If not that's also fine, but I think that'd be a much preferable approach

Reply:
I actually did it this way since from what I understand, stream based approaches are much less performant. As these operations involve a few nested loops (albeit at a small scale), I thought it'd be good to save execution time where possible.

This might be a case of premature optimisation though... Can definitely change it to be stream-based.

Reply:
While this is probably true (I've never actually seen any real-world benchmarks about this, just a lot of people shouting loud warnings), there is no way those performance differences would ever be relevant here and having less code (and clearer code, too) is definitely the way to go here